.github/workflows/doc-build.yml

```diff
@@ -19,30 +19,79 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v3
 
-      - name: Prepare
+      - name: Update system packages
         run: |
-          apt-get update && apt-get install -y zip
-          # Reuse built SAGE_LOCAL contained in the Docker image
-          ./bootstrap
-          ./configure --enable-build-as-root --prefix=/sage/local --with-sage-venv --enable-download-from-upstream-url
+          apt-get update && apt-get install -y git zip
 
-      - name: Build
-        run: make doc-html
+      - name: Add prebuilt tree as a worktree
+        id: worktree
+        run: |
+          set -ex
+          git config --global user.email "ci-sage@example.com"
+          git config --global user.name "Build & Test workflow"
+          git config --global --add safe.directory $(pwd)
+          # If actions/checkout downloaded our source tree using the GitHub REST API
+          # instead of with git (because do not have git installed in our image),
+          # we first make the source tree a repo.
+          if [ ! -d .git ]; then git init && git add -A && git commit --quiet -m "new"; fi
+          # Tag this state of the source tree "new". This is what we want to build and test.
+          git tag -f new
+          # Our container image contains a source tree in /sage with a full build of Sage.
+          # But /sage is not a git repository.
+          # We make /sage a worktree whose index is at tag "new".
+          # We then commit the current sources and set the tag "old". (This keeps all mtimes unchanged.)
+          # Then we update worktree and index with "git reset --hard new".
+          # (This keeps mtimes of unchanged files unchanged and mtimes of changed files newer than unchanged files.)
+          # Finally we reset the index to "old". (This keeps all mtimes unchanged.)
+          # The changed files now show up as uncommitted changes.
+          git worktree add --detach worktree-image
+          rm -rf /sage/.git && mv worktree-image/.git /sage/
+          rm -rf worktree-image && ln -s /sage worktree-image
+          if [ ! -f worktree-image/.gitignore ]; then cp .gitignore worktree-image/; fi
+          (cd worktree-image && git add -A && git commit --allow-empty -m "old" -a && git tag -f old && git reset --hard new && git reset old)
+          # Keep track of changes to built HTML
+          (cd /sage/local/share/doc/sage/html/en && find . -name "*.html" | xargs sed -i '/class="sidebar-brand-text"/s/Sage [0-9a-z.]* /Sage dev /'; git init && (echo ".buildinfo"; echo ".inv") > .gitignore; git add -A && git commit -m "old")
+
+      - name: Incremental build and docbuild
+        id: incremental
+        run: |
+          # Now re-bootstrap and build. The build is incremental because we were careful with the timestamps.
```

Comment:
I thought the incremental docbuilds are not stable right now?

Reply:
> The changes https://deploy-preview-35652--sagemath-tobias.netlify.app/changes.html seem to list too many files. 

There was a global change (a mathjax macro added) since the baseline that changed all files

> Could you also add the actual diff there? That would be handy in many instances.

There is a bit of the diff shown in the workflow; not enough?

Reply:
I don't understand, isn't the baseline the current dev branch?

> There is a bit of the diff shown in the workflow; not enough?

Yes, that's enough. Just put it in the changes.html, or if you feel like it use https://diff2html.xyz/

Reply:
> I don't understand, isn't the baseline the current dev branch?

Normally yes, but a bug (#35737) prevented the image to be built on the last beta.

Reply:
Okay, thanks that makes sense!

Reply:
Full diff is now packaged up; as of 9f30a8d